cannot import node built-ins in vite, even in server-only code

### Environment

- Operating System: `Linux`
- Node Version:     `v16.13.0`
- Nuxt Version:     `3.0.0-27288761.d3fb512`
- Package Manager:  `npm@8.1.0`
- Bundler:          `Vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

Create and initialize a Nuxt 3 project:

```sh
$ npx nuxi init nuxt3-app
$ cd nuxt3-app
$ npm install
```

Then, install the [uuid](https://www.npmjs.com/package/uuid) package, import it in the app.vue file, and start the dev server:

```sh
$ npm install uuid
$ vim app.vue
$ cat app.vue
<script setup>
import { v4 as uuidv4 } from 'uuid';
</script>

<template>
  <div>
    <NuxtWelcome />
  </div>
</template>
$ npm run dev
```

Accessing localhost:3000 from the browser causes a 500 error.

### Describe the bug

The dev server crashes if the uuid package is imported in a script block.

Note: If you change `<script setup>` to `<script setup lang="ts">`, it seems like the server operates without an error.

### Additional context

_No response_

### Logs

```shell
Nuxt CLI v3.0.0-27288761.d3fb512                                                                                                                                                                                                    14:07:18
                                                                                                                                                                                                                                    14:07:18
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.1.60:3000/
  > Network:  http://172.17.0.1:3000/

ℹ Vite warmed up in 353ms                                                                                                                                                                                                           14:07:20
✔ Vite server built in 699ms                                                                                                                                                                                                        14:07:20
✔ Nitro built in 141 ms                                                                                                                                                                                                             14:07:20
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/me/dev/temp/nuxt3-app/crypto' imported from /home/me/dev/temp/nuxt3-app/.nuxt/dist/server/server.mjs
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ESMLoader.import (node:internal/modules/esm/loader:276:22)
    at importModuleDynamically (node:internal/modules/esm/translators:111:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at $id_da2f073e (file:///home/shf0811/dev/temp/nuxt3-app/.nuxt/dist/server/server.mjs:2009:220) {
  code: 'ERR_MODULE_NOT_FOUND'
}
[vite dev] Error loading external "crypto".
  at file://./.nuxt/dist/server/server.mjs:2009:252  
  at async __instantiateModule__ (file://./.nuxt/dist/server/server.mjs:2495:3)
[vite dev] Error loading external "crypto".
  at file://./.nuxt/dist/server/server.mjs:2009:252  
  at async __instantiateModule__ (file://./.nuxt/dist/server/server.mjs:2495:3)
```

--- a/packages/vite/src/dev-bundler.ts
+++ b/packages/vite/src/dev-bundler.ts
@@ -1,5 +1,6 @@
 import { pathToFileURL } from 'url'
 import { existsSync } from 'fs'
+import { builtinModules } from 'module'
 import { resolve } from 'pathe'
 import * as vite from 'vite'
 import { ExternalsOptions, isExternal as _isExternal, ExternalsDefaults } from 'externality'
@@ -73,8 +74,11 @@ async function transformRequest (opts: TransformOptions, id: string) {
   // Remove for externals
   const withoutVersionQuery = id.replace(/\?v=\w+$/, '')
   if (await isExternal(opts, withoutVersionQuery)) {
+    const path = builtinModules.includes(withoutVersionQuery.split('node:').pop())
+      ? withoutVersionQuery
+      : pathToFileURL(withoutVersionQuery)
     return {
-      code: `(global, exports, importMeta, ssrImport, ssrDynamicImport, ssrExportAll) => import('${(pathToFileURL(withoutVersionQuery))}').then(r => { exports.default = r.default; ssrExportAll(r) }).catch(e => { console.error(e); throw new Error('[vite dev] Error loading external "${id}".') })`,
+      code: `(global, exports, importMeta, ssrImport, ssrDynamicImport, ssrExportAll) => import('${path}').then(r => { exports.default = r.default; ssrExportAll(r) }).catch(e => { console.error(e); throw new Error('[vite dev] Error loading external "${id}".') })`,
       deps: [],
       dynamicDeps: []
     }

